generator/src/main/java/com/scottlogic/deg/generator/generation/StandardFieldValueSourceEvaluator.java

```diff
@@ -108,20 +108,42 @@ private FieldValueSource getNumericSource(FieldSpec fieldSpec) {
             ? new NumericRestrictions()
             : fieldSpec.getNumericRestrictions();
 
-        int numericScale = fieldSpec.getGranularityRestrictions() != null
-            ? fieldSpec.getGranularityRestrictions().getNumericScale()
-            : 0;
+        int numericScale = getNumericScale(restrictions, fieldSpec);
 
-        if (numericScale == 0) {
+        if ((restrictions.min == null || restrictions.max == null) || isIntegerFieldValue(restrictions, numericScale)) {
             return new IntegerFieldValueSource(
                 restrictions,
                 getBlacklist(fieldSpec));
-        } else {
-            return new RealNumberFieldValueSource(
-                restrictions,
-                getBlacklist(fieldSpec),
-                numericScale);
         }
+
+        return new RealNumberFieldValueSource(
+            restrictions,
+            getBlacklist(fieldSpec),
+            numericScale);
+    }
+
+    private int getNumericScale(NumericRestrictions numericRestrictions, FieldSpec fieldSpec) {
```

Comment:
Not introduced by your PR but can we rename NumericRestrictions to NumericRestriction? The plural is confusing me here. Plural class names in general seem weird to me

Reply:
Based on the fact we have quite a few other classes with similar names including other restrictions this seems like a wider task if this is something we want to do

Reply:
This type can represent 2 restrictions over numeric values, the upper and/or lower bound, as such i think pluralisation here is ok.